lib/puppet/util/log/destinations.rb

```diff
@@ -232,6 +233,16 @@ def to_native(level)
 
   def handle(msg)
     native_type, native_id = to_native(msg.level)
+	
+	stringified_msg = msg.message.to_s
+	
+	if stringified_msg.length > self.class::EVENTLOG_CHARACTER_LIMIT
+		warning = "Message exceeds character length limit, truncating..."
```

Comment:
This will append the warning to the log without any delimiter, so if we cut off somewhere it could look like:
> ...this  is the end of a very lonMessage exceeds character length limit, truncating..."

Can you insert a delimiter? I think maybe swap the ellipsis to prepend and terminate with a period?  I.e.,
> ...this is the end of a very lon...Message exceeds character length limit, truncating."